### ISSUE TITLE
[rx-51] wrong keymap in X

### ISSUE THREAD

**Issue by pavelmachek** [postmarketOS/pmbootstrap#701]
It seems we have wrong keymap in X: blue key does not work, so symbols can not be accessed. `/usr/share/X11/xkb/symbols/nokia_vndr/rx-51` exists and seems to contain something reasonable.

**Comment by ollieparanoid** [comment]
I found [this repo](https://github.com/archlinuxarm-n900/xkeyboard-config-n900-git) which basically patches the keyboard layout from upstream as I understand.

Maybe you could make a `xkeyboard-nokia-rx51` package for starters, which works similarly (and to make sure that it still works), and then try to get these patches upstreamed so we can drop the package again.

**Comment by craftyguy** [comment]
Wow nice find @ollieparanoid , I was searching for some way to implement this a while back but wasn't able to find this project.

**Comment by drebrez** [comment]
@ollieparanoid @craftyguy I looked at the patches in that repo and they change the X11 keyboard layout to be like in this picture: 
![](https://wiki.postmarketos.org/images/e/e1/N900_keymap_us.jpg)
which I created following the [rx51_us.map](https://github.com/postmarketOS/pmbootstrap/blob/master/aports/device/device-nokia-n900/keymaps/rx51_us.map) file 👍

**Comment by ollieparanoid** [comment]
@drebrez: if I understood you right this can be closed now. Please re-open if there's still something left to do.

**Comment by drebrez** [comment]
@ollieparanoid I just mentioned that those patches are for setting the X11 keyboard layout as the layout we have for the console.
We actually need to apply those patches somewhere, because right now in X11 the layout doesn't have all the special characters. 
We also need a X11 layout for other languages, like the `ch_de` which uses the arrow keys differently

**Comment by drebrez** [comment]
@pavelmachek @ollieparanoid @MartijnBraam  I'm trying to have the keymap in X exactly like those we have in the console (which I have created some explaining images in the wiki https://wiki.postmarketos.org/wiki/Nokia_N900#Keyboard_layout), does it make sense to have it the same?

Then I'm not sure which solution is the best between:

1. Apply some patch files (like in the repo @ollieparanoid found) to the original `/usr/share/X11/xkb/symbols/nokia_vndr/rx-51` file
2. Create a whole new keyboard layout (`pmos_vndr/n900`?) and add a layout entry to the `/usr/share/X11/xkb/rules/evdev.xml` file

what you think?

**Comment by ollieparanoid** [comment]
> does it make sense to have it the same?

Yes, in my opinion it does. It's not intuitive when the keyboard layout is suddenly different when switching from tty0 to X11 or vice versa.

> Then I'm not sure which solution is the best between:

In both cases, we would need to fork and override Alpine's `xkeyboard-config` package (it provides both files you suggested to change).

I would go with "1." and once the modified keymaps are tried and tested for a while, I'd recommend to upstream them to the official sources of `xkeyboard-config` so everybody benefits from it. Then we could also remove our forked package again.
